### PR TITLE
Switch jobstats to streaming impl

### PR DESCRIPTION
**Pull request by jgrund**
When querying a large amount of jobstats (25k+), we see a slowdown in response time and a large spike in memory usage.

This is because we store the whole result of `lctl get_param` for jobstats in memory and perform a few clones on the data.

Instead, we should spawn the `lctl get_param` call and stream the stdout line by line (with a small in memory buffer and bounded channel). This stream can then be transformed and emitted as jobstats data in prometheus exposition format directly over socket to the caller.

This will allow constant memory usage (up to filled buffer / channel size) and backpressure due to how fast the client can read / how fast we can process lines.

- Switch jobstats pipeline to streaming impl
- Merge stream with existing stats and push over network socket
- Use compact_string to try and avoid heap allocations

### CODE REVIEW

**Comment by RDruon** [comment]
Scraping the exporter from a prometheus server works as expected with latest commit:
```
[root@c175 prometheus]# promtool query instant http://localhost:9090/ "count(count by (jobid) (lustre_job_stats_total))"
{} => 36603 @[1723043864.197]
```
The server above is scraping a single exporter.

File: lustrefs-exporter/src/jobstats.rs [hunk 1]
```diff
@@ -46,486 +51,325 @@ static WRITE_BYTES: Metric = Metric {
...
+                }
+            }
+        }
+
+        if let State::TargetJobStats(target, job, stats) = state {
+            render_stat(tx.clone(), &target, job, stats)?;
+        }
+
+        Ok(())
+    });
```
Comment (jgrund) [inline lustrefs-exporter/src/jobstats.rs:124]: need to chain a combinator here to catch errors and log them at debug level

File: lustrefs-exporter/src/jobstats.rs [hunk 2]
```diff
@@ -70,57 +70,64 @@ pub fn jobstats_stream<R: BufRead + std::marker::Send + 'static>(
 ) -> (JoinHandle<Result<(), Error>>, Receiver<CompactString>) {
     let (tx, rx) = mpsc::channel(200);
 
-    let x = tokio::task::spawn_blocking(move || {
-        let mut state = State::Empty;
+    let x = tokio::task::spawn(async move {
+        tokio::task::spawn_blocking(move || {
+            let mut state = State::Empty;
```
Comment (RDruon) [inline lustrefs-exporter/src/jobstats.rs:75]: It allows to return a `JoinHandle` and await the blocking task in it.
@jgrund if you have a better idea

Comment (jgrund) [reply lustrefs-exporter/src/jobstats.rs:75]: I'll push something, one sec...

**Pull request merged**